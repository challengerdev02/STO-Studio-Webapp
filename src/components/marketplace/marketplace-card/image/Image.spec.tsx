import {
  cleanup,
  queryByAttribute,
  queryByRole,
  render,
  waitFor,
} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { CustomStyledImage, generateStackedImage } from './index';

describe('Container', function () {
  afterEach(cleanup);
  const StackImageReact = ({ urlList }: any) => (
    <>{generateStackedImage(urlList ?? [])}</>
  );
  let props = {
    width: '256px',
    height: '256px',

    hoverable: true,
    urlList: [
      'https://robohash.org/962440efdb9e25e8ff4360a7037acfb4?set=set4&bgset=&size=400x400',
      'https://robohash.org/887f430666b7d04d90b82a4b988df961?set=set4&bgset=&size=400x400',
      'https://robohash.org/15b76d2ec3afc53e3af6c4a316090c94?set=set4&bgset=&size=400x400',
    ],
  };

  const setup = (extraProps: Record<string, any> = {}) => {
    const utils = render(<CustomStyledImage {...props} {...extraProps} />);
    return {
      ...utils,
    };
  };

  const setupImage = (extraProps: Record<string, any> = {}) => {
    const StackImageContent = ({ ...props }: { urlList: string[] }) => {
      return (
        <div
          role={'stack-container'}
          style={{ display: 'flex', position: 'relative' }}
        >
          {generateStackedImage(props.urlList)}
        </div>
      );
    };
    const utils = render(<StackImageContent {...props} {...extraProps} />);
    return {
      ...utils,
    };
  };

  beforeEach(() => {
    Object.defineProperty(window, 'matchMedia', {
      writable: true,
      value: jest.fn().mockImplementation((query) => ({
        matches: false,
        media: query,
        onchange: null,
        addListener: jest.fn(), // deprecated
        removeListener: jest.fn(), // deprecated
        addEventListener: jest.fn(),
        removeEventListener: jest.fn(),
        dispatchEvent: jest.fn(),
      })),
    });
  });

  it('should match with snapshot', () => {
    // @ts-ignore
    const { container } = setup();
    expect(container).toMatchSnapshot();
  });

  it('should have Image "This is a test Image"', () => {
    // @ts-ignore
    const { container } = setup({
      src: 'https://robohash.org/962440efdb9e25e8ff4360a7037acfb4?set=set4&bgset=&size=400x400',
    });
    expect(
      queryByAttribute(
        'src',
        document.body,
        'https://robohash.org/962440efdb9e25e8ff4360a7037acfb4?set=set4&bgset=&size=400x400'
      )
    ).toBeInTheDocument();
  });

  it('should match with snapshot "generateStackedImage"', async () => {
    // @ts-ignore
    const { container } = setupImage();
    await waitFor(() => {
      expect(container).toMatchSnapshot();
    });
  });

  it('should have Image "This is a test Stacked Image"', () => {
    // @ts-ignore
    const { container } = setupImage();
    expect(queryByRole(document.body, 'stack-container')).toBeInTheDocument();
  });
});
