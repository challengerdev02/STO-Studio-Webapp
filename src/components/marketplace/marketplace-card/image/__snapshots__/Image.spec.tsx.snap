// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container should match with snapshot "generateStackedImage" 1`] = `
<div>
  <div
    role="stack-container"
    style="display: flex; position: relative;"
  >
    <div
      class="sc-dkPtRN sc-hKwDye kToXgV rBElr stacked-img"
      height="24px"
      role="stackImages"
      width="24px"
    >
      <img
        alt="stack-image"
        class="sc-gKclnd fRPrfE"
        src="https://robohash.org/962440efdb9e25e8ff4360a7037acfb4?set=set4&bgset=&size=400x400"
        width="100%"
      />
    </div>
    <div
      class="sc-dkPtRN sc-hKwDye kToXgV kFSgPO stacked-img"
      height="24px"
      role="stackImages"
      width="24px"
    >
      <img
        alt="stack-image"
        class="sc-gKclnd fRPrfE"
        src="https://robohash.org/887f430666b7d04d90b82a4b988df961?set=set4&bgset=&size=400x400"
        width="100%"
      />
    </div>
    <div
      class="sc-dkPtRN sc-hKwDye kToXgV heHuFZ stacked-img"
      height="24px"
      role="stackImages"
      width="24px"
    >
      <img
        alt="stack-image"
        class="sc-gKclnd fRPrfE"
        src="https://robohash.org/15b76d2ec3afc53e3af6c4a316090c94?set=set4&bgset=&size=400x400"
        width="100%"
      />
    </div>
  </div>
</div>
`;

exports[`Container should match with snapshot 1`] = `
<div>
  <img
    class="sc-gKclnd dXTNPF"
    height="256px"
    width="256px"
  />
</div>
`;
